data/README.md

```diff
@@ -0,0 +1,100 @@
+ingesting, processing, and aggregating data so that it can be displayed to users
+or made available to data scientists.
+
+*Data Science*: the practice of using statistics, machine learning, and other
+tools to analyze data to discover trends and truths that can be used to provide
+business intelligence.
+
+*Batch Processing*: processing large amounts of data at once. This is acceptable
+for smaller amounts of data and can be simpler in terms of engineering and
+deployment. Some batch processes can also be useful for "recomputing the world"
+when you want to analyze existing data in a new way.
+
+*Data Streaming*: processing data in small chunks, one at a time, rather than
+processing all data at once. Streaming is necessary for processing infinite
+event streams. It's also useful for processing large amounts of data, because it
+prevents memory overflows during processing and makes it easier to process data
+in a distributed manner or real-time manner.
+
+*Real-time*: analyzing data and delivering results simultaneously so that stream
+output is always visible. For example, real-time analytics will mean that the
+system is constantly processing events (clicks, purchases, etc) and displaying
+the latest results in a user interface.
+
+*Parallel Processing*: performing multiple tasks at the same time, for example
+on different cores or processors. Parallel processing is necessary in order to
+perform more than one computation at once, such as parsing or aggregation.
+
+*Concurrent Processing*: managing multiple ongoing tasks at once without
+necessarily processing more than one task in the same exact moment. Concurrent
+processing is required in order to perform more than one effect at once, such as
+waiting for multiple network requests to complete.
+
+*Constant Memory*: processing a stream where the amount of memory required does
+not increase with the size of the stream.
+
+*At Least Once Delivery*: a guarantee that a given message will delivered at
+least once, but may be delivered more than once. This is achieved in Kafka by
+committing an offset after it's been fully processed and in RabbitMQ by
+acknowledging a message after fully processing it.
+
+*At Most Once Delivery*: a guarantee that a message will never be delivered more
+than once, but may not be delivered at all. This is achieved in Kafka by
+committing an offset before fully processing a message and in RabbitMQ by
+acknowledging a message before fully processing it.
+
+*Distributed Data Processing*: breaking up data into partitions so that large
+amounts of data can be processed by many machines simultaneously.
+
+*Cluster*: several computers (or virtual machines) grouped together to perform a
+single task.
+
+*Scala*: a programming language (like Ruby, Python, or JavaScript) which is fast
+and has become popular for data-focused tasks. Scala runs on the Java Virtual
+Machine, which is a high-performance engine for running languages like Scala
```

Comment:
You are being very kind to the JVM here.

Reply:
It seemed impolite to mention type erasure or embarrassing start-up latency.